usaspending_api/disaster/v2/views/agency/count.py

```diff
@@ -17,18 +19,28 @@ class AgencyCountViewSet(CountBase):
 
     @cache_response()
     def post(self, request: Request) -> Response:
-        filters = [
-            Q(treasury_account__funding_toptier_agency=OuterRef("pk")),
-            self.is_in_provided_def_codes(),
-            self.all_closed_defc_submissions,
-            self.is_non_zero_object_class_cpe(),
-        ]
-        count = (
-            ToptierAgency.objects.annotate(
-                include=Exists(FinancialAccountsByProgramActivityObjectClass.objects.filter(*filters).values("pk"))
+        if self.award_type_codes:
+            filters = [
+                self.is_in_provided_def_codes(),
+                self.all_closed_defc_submissions,
+                self.is_non_zero_award_cpe(),
```

Comment:
Looking at the code for `DisasterBase.is_non_zero_award_cpe()`, it looks wrong:

```
    def is_non_zero_award_cpe(self):
        return Q(
            Q(obligations_incurred_total_by_award_cpe__gt=0)
            | Q(obligations_incurred_total_by_award_cpe__lt=0)
            | Q(gross_outlays_delivered_orders_paid_total_cpe__gt=0)
            | Q(gross_outlays_delivered_orders_paid_total_cpe__lt=0)
        )
```

If this is indeed at the _award_ metrics of account data -- i.e. File C / `FinancialAccountsByAwards` -- then it should be using these fields to yield sums, and therefore these fields should be checked as non-zero:
- obligations: `FinancialAccountsByAwards.transaction_obligated_amount`
- outlay: `FinancialAccountsByAwards.gross_outlay_amount_by_award_cpe`